core/src/test/java/feign/RequestTemplateTest.java

```diff
@@ -362,4 +362,18 @@ public void encodedQueryWithUnsafeCharactersMixedWithUnencoded() throws Exceptio
     assertThat(template.queries()).doesNotContain(entry("params[]", asList("not encoded")));
     assertThat(template.queries()).contains(entry("params[]", asList("encoded")));
   }
+  
+  @Test
+  public void appendDirectlytest() throws Exception {
+    String url = "http://test.feign.com/query?params=%7b%22int%22%3a1%2c%22string%22%3a%22str%22%7d";
```

Comment:
comment why we care. ex people won't necessarily be able to understand that this is escaped json (if it is)